src/main/java/com/legacyminecraft/poseidon/util/ServerLogRotator.java

```diff
@@ -0,0 +1,101 @@
+package com.legacyminecraft.poseidon.util;
+
+import org.bukkit.Bukkit;
+import com.legacyminecraft.poseidon.PoseidonPlugin;
+
+import java.io.File;
+import java.io.PrintWriter;
+import java.nio.file.Files;
+import java.time.Duration;
+import java.time.LocalDateTime;
+import java.time.ZonedDateTime;
+import java.time.format.DateTimeFormatter;
+import java.util.concurrent.TimeUnit;
+import java.util.logging.Logger;
+import java.util.logging.*;
+
+/**
+ * The ServerLogRotator class handles the rotation of server logs.
+ * It archives the current log file and creates a new one every day at midnight.
+ */
+public class ServerLogRotator {
+
+    private final String latestLogFileName;
+    private final Logger a;
+
+    /**
+     * Constructor for the ServerLogRotator class.
+     *
+     * @param latestLogFileName The name of the latest log file. This file will be archived by the log rotation task.
+     */
+    public ServerLogRotator(String latestLogFileName) {
+        this.latestLogFileName = latestLogFileName;
+        this.a = Logger.getLogger("Minecraft");
+    }
+    // end of constructor
+
+    /**
+     * The archiveLog method is responsible for archiving the current log file.
+     * It first checks if the latest log file exists. If it does, it reads its contents,
+     * writes the contents to a new file named with the current date, and then clears the latest log file.
+     * If the latest log file does not exist, it logs a message and does nothing.
+     * If any exception occurs during the process, it logs the exception message.
+     */
+    private void archiveLog() {
+        try {
+
+            this.a.log(Level.INFO, "Archiving contents of latest.log to a new file!");
```

Comment:
only small comment i have left is to rename `a` for readability and also you can say "a" instead of "this.a" since the variable is belonging to the class

Reply:
Fixed.